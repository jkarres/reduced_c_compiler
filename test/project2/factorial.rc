
function : int factorial(int n) {
  if (n == 0 || n == 1) {
    return 1;
  }

  return n * factorial(n-1);

}

function : void main() { 
  cout << "0! is " << factorial(0) << endl;
  cout << "1! is " << factorial(1) << endl;
  cout << "2! is " << factorial(2) << endl;
  cout << "3! is " << factorial(3) << endl;
  cout << "4! is " << factorial(4) << endl;
  cout << "5! is " << factorial(5) << endl;
}
