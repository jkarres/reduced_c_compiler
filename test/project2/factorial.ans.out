0! is 1
1! is 1
2! is 2
3! is 6
4! is 24
5! is 120
